rvgo/fast/vm.go

```diff
@@ -894,7 +875,9 @@ func (inst *InstrumentedState) riscvStep() (outErr error) {
 			revertWithCode(riscv.ErrBadAMOSize, fmt.Errorf("bad AMO size: %d", size))
 		}
 		addr := getRegister(rs1)
-		// TODO check if addr is aligned
+		if addr&3 != 0 { // quick addr alignment check
```

Comment:
See other comment on removing the ability to write across multiple pages in `SetAligned`.